modules/xfem/test/tests/crack_tip_enrichement/penny_crack_3d.i

```diff
@@ -0,0 +1,367 @@
+[GlobalParams]
+  order = FIRST
+  family = LAGRANGE
+  crack_front_definition = crack_front
+  enrichment_displacements = 'enrich1_x enrich2_x enrich3_x enrich4_x enrich1_y enrich2_y enrich3_y enrich4_y enrich1_z enrich2_z enrich3_z enrich4_z'
+  cut_off_radius = 0.3
+[]
+
+[XFEM]
+  geometric_cut_userobjects = 'circle_cut_uo'
+  qrule = volfrac
+  output_cut_plane = true
+[]
+
+[UserObjects]
+  [./circle_cut_uo]
+    type = CircleCutUserObject
+    cut_data = '0 0 0
+                0.5 0 0
+                0 0.5 0'
+  [../]
+  [./crack_front]
+    type = CrackFrontDefinition
+    crack_direction_method = CurvedCrackFront
+    crack_front_points = '0.500000000000000                   0                   0
+                          0.000000000000000   0.500000000000000                   0
+                         -0.500000000000000   0.000000000000000                   0
+                         -0.000000000000000  -0.500000000000000                   0'
+  [../]
+[]
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 3
+  nx = 9
+  ny = 9
+  nz = 3
+  xmin = -1.0
+  xmax = 1.0
+  ymin = -1.0
+  ymax = 1.0
+  zmin = -0.75
+  zmax = 0.75
+  elem_type = HEX8
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[MeshModifiers]
+  [./all_node]
+    type = BoundingBoxNodeSet
+    new_boundary = 'all'
+    top_right = '1 1 1'
+    bottom_left = '-1 -1 -1'
+  [../]
+[]
+
+[Variables]
+  [./disp_x]
+  [../]
+  [./disp_y]
+  [../]
+  [./disp_z]
+  [../]
+  [./enrich1_x]
```

Comment:
Why not use the Action to set all this up?

Reply:
Done